modules/indigo/module/inc/indigo/of_state_manager.h

```diff
@@ -141,13 +141,34 @@ indigo_core_stats_get(uint32_t *total_flows,
  * called when a gentable_entry_add message is received for a key that
  * already exists in the table.
  *
+ * Async operation support:
+ * add4()/modify4()/del4() are used for async operations.
+ * These new APIs can return INDIGO_ERROR_PENDING to state manager
+ * if the operations are async.
+ * The state manager provides a resume function,
+ * indigo_core_gentable_resume(), to complete the remaining works.
```

Comment:
How about something like this?
"In the event that PENDING is returned by the add4/modify4/del4 APIs, the resume function indigo_core_gentable_resume() must be called by the driver either to complete any remaining work if the asynchronous operation is successfully completed by the lower-level driver, or to clean up operational state if the asynchronous operation fails in lower-level driver code.  In order to provide sufficient context to the resume function, the add4/modify4/del4 APIs take a void* context as their first parameter.  This void* value should be passed back to the resume function when the lower-level driver code finally completes (either with success or failure)."